connectors/sources/jira.py

```diff
@@ -0,0 +1,493 @@
+                "type": "str",
+            },
+            "service_account_id": {
+                "value": "me@example.com",
+                "label": "Jira Cloud service account id",
+                "type": "str",
+            },
+            "api_token": {
+                "value": "abc#123",
+                "label": "Jira Cloud API token",
+                "type": "str",
+            },
+            "host_url": {
+                "value": "http://127.0.0.1:8080",
+                "label": "Jira host url",
+                "type": "str",
+            },
+            "ssl_enabled": {
+                "value": False,
+                "label": "Enable SSL verification (true/false)",
+                "type": "bool",
+            },
+            "ssl_ca": {
+                "value": "",
+                "label": "SSL certificate",
+                "type": "str",
+            },
+            "enable_content_extraction": {
+                "value": True,
+                "label": "Enable content extraction (true/false)",
+                "type": "bool",
+            },
+            "retry_count": {
+                "value": 3,
+                "label": "Maximum retries for failed requests",
+                "type": "int",
+            },
+            "concurrent_downloads": {
+                "value": 50,
+                "label": "Number of concurrent downloads for fetching attachment content",
+                "type": "int",
+            },
+        }
+
+    def tweak_bulk_options(self, options):
+        """Tweak bulk options as per concurrent downloads support by jira
+
+        Args:
+            options (dictionary): Config bulker options
+        """
+        options["concurrent_downloads"] = self.concurrent_downloads
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+
+        Raises:
+            Exception: Configured keys can't be empty
+        """
+        logger.info("Validating Jira Configuration")
+        connection_fields = (
+            ["host_url", "service_account_id", "api_token"]
+            if self.is_cloud
+            else ["host_url", "username", "password"]
+        )
+
+        default_config = self.get_default_configuration()
+
+        if empty_connection_fields := [
+            default_config[field]["label"]
+            for field in connection_fields
+            if self.configuration[field] == ""
+        ]:
+            raise Exception(
+                f"Configured keys: {empty_connection_fields} can't be empty."
+            )
+
+        if self.ssl_enabled and self.certificate == "":
+            raise Exception("SSL certificate must be configured.")
+
+        if self.concurrent_downloads > MAX_CONCURRENT_DOWNLOADS:
+            raise Exception(
+                f"Configured concurrent downloads can't be set more than {MAX_CONCURRENT_DOWNLOADS}."
+            )
+
+    def _generate_session(self):
```

Comment:
I would really like to see the same pattern we've used for the Confluence connector (see: https://github.com/elastic/connectors-python/pull/568/files), i.e. creating a `JiraClient` class, which encapsulates the interaction with the server/cloud instance. Makes the connectors way more consistent.

Reply:
Sure, however the testing cycle has been completed and it would take quite some efforts from the QA end to re-test it with new changes, so I strongly recommend we raise a GitHub issue and cover it as an enhancement PR. Let me know your thoughts regarding the same.